### PR TITLE
Git guidelines draft

### PR DESCRIPTION
**Pull request by D4N**
This is the promised draft of the git guidelines which I promised to write and promptly forgot to upload. Thanks to @frli8848 for reminding me!

### CODE REVIEW

File: git_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+- For each release a branch of the form `$major.$minor` should be created to
+  store backported changes. It should be branched of from `master` at the commit
+  which was tagged with `v$major.$minor`.
+
+- All other branches are development branches for pull requests, experiments,
+  etc. They should be deleted once the pull request got merged or the branch is
+  no longer useful.
+
+- Exiv2 team members should create branches for pull requests in the main
```
Comment (piponazo) [inline git_guidelines.md]: Should we always follow this pattern? At this moment, I am always using my own fork to collaborate. I think it is a protection mechanism against stupid human errors.

But I agree with you in which we could use branches in the main repository when collaboration is needed.

**Review by piponazo — APPROVED** [review]
Great work @D4N! I'm sure this will help to lot of people who want to contribute to the project

File: git_guidelines.md (hunk 1 above)
Comment (D4N) [reply git_guidelines.md]: Actually I am also using my own fork all the time, so I'll reword this a bit. Branches in the main repo probably only make sense if we are going to collaborate on something.

**Comment by D4N** [comment]
@frli8848 Could you take a look at the file too and tell me what you think? 

During the team meeting we also discussed whether some convenience scripts for git could help (e.g. a script to create a new branch automatically, prepare the remotes, etc). What's your opinion on that?

**Comment by frli8848** [comment]
@D4N Yes I can have a look at it (during next week). And, I think convenience scripts can be good. Perhaps one script for those who fork the repo and one script for exiv2 admins (that work on branches in the main/upstream repo).  Also, I think it would be good if the scripts print info of each command it performs.

**Comment by D4N** [comment]
Thanks, I'll try to create some scripts also in that time frame.

**Comment by frli8848** [comment]
@D4N I finally got some time to check this. Here is my comments [I number them "1)", "2)",  etc]:

1) The file name should be upper case (then the link from CONTRIBUTING.md will work)

2) Add a TOC (I can do this later if you want)

3) Add section numbers (I had to do this manually on the CONTRIBUTING.md file). 

4) Add a short introduction. Something like:

This document presents some guidelines for working with the git (distributed) version control system on the Exiv2 project. The objective is to keep the Exiv2 repository ....  This document also treat how releases are tagged, how to backport critical bugfixes, and some general hints on how to use git.

5) Section "Commit messages", line 7: Add a short note on what travis is.

6) Section "Keeping the history linear", line 111. Add a short note on what cherry-picking and bisecting is, and/or give a pointer(s) to where the reader can find info on what it is.

7) Section "Branches and tags". Do you only use two version numbers? Or is "patch" missing? Eg., `v$major.$minor.$patch`

Otherwise, I think this is a very nice read Dan!

**Review by clanmills — APPROVED** [review]
Wonderful.

**Pull request merged**